pkg/clients/ec2/securitygroup.go

```diff
@@ -52,13 +52,58 @@ func IsRuleAlreadyExistsErr(err error) bool {
 	return errors.As(err, &awsErr) && awsErr.ErrorCode() == InvalidPermissionDuplicate
 }
 
+// BuildFromEC2Permissions converts ec2 Permissions to our API's format.
+func BuildFromEC2Permissions(in []ec2types.IpPermission) []v1beta1.IPPermission {
+	if len(in) == 0 {
+		return nil
+	}
+	permissions := make([]v1beta1.IPPermission, len(in))
+	for i, p := range in {
+		ipPerm := v1beta1.IPPermission{
+			FromPort:   p.FromPort,
+			IPProtocol: aws.StringValue(p.IpProtocol),
+			ToPort:     p.ToPort,
+		}
+		for _, c := range p.IpRanges {
+			ipPerm.IPRanges = append(ipPerm.IPRanges, v1beta1.IPRange{
+				CIDRIP:      aws.StringValue(c.CidrIp),
+				Description: c.Description,
+			})
+		}
+		for _, c := range p.Ipv6Ranges {
+			ipPerm.IPv6Ranges = append(ipPerm.IPv6Ranges, v1beta1.IPv6Range{
+				CIDRIPv6:    aws.StringValue(c.CidrIpv6),
+				Description: c.Description,
+			})
+		}
+		for _, c := range p.PrefixListIds {
+			ipPerm.PrefixListIDs = append(ipPerm.PrefixListIDs, v1beta1.PrefixListID{
+				Description:  c.Description,
+				PrefixListID: aws.StringValue(c.PrefixListId),
+			})
+		}
+		for _, c := range p.UserIdGroupPairs {
+			ipPerm.UserIDGroupPairs = append(ipPerm.UserIDGroupPairs, v1beta1.UserIDGroupPair{
+				Description:            c.Description,
+				GroupID:                c.GroupId,
+				GroupName:              c.GroupName,
+				UserID:                 c.UserId,
+				VPCID:                  c.VpcId,
+				VPCPeeringConnectionID: c.VpcPeeringConnectionId,
+			})
+		}
+		permissions[i] = ipPerm
+	}
+	return permissions
+}
```

Comment:
Reviewer note: This is pretty much a copy and paste of `GenerateEC2Permissions` but inverted.